tutorials/preprocessing/plot_40_artifact_correction_ica.py

```diff
@@ -1,307 +1,513 @@
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 #
-# Pay attention to the labels, a customized read-out of the
-# ``mne.preprocessing.ICA.labels_``:
-print(ica.labels_)
+# It may have seemed easy to review the plots and manually select which ICs to
+# exclude, but when processing dozens or hundreds of subjects this can become
+# a tedious, rate-limiting step in the analysis pipeline. One alternative is to
+# use dedicated EOG or ECG sensors as a "pattern" to check the ICs against, and
+# automatically mark for exclusion any ICs that match the EOG/ECG pattern. Here
+# we'll use :meth:`~mne.preprocessing.ICA.find_bads_eog` to automatically find
+# the ICs that best match the EOG signal, then use
+# :meth:`~mne.preprocessing.ICA.plot_scores` along with our other plotting
+# functions to see which ICs it picked. We'll start by resetting
+# ``ica.exclude`` back to an empty list:
+
+ica.exclude = []
+# find which ICs match the EOG pattern
+eog_indices, eog_scores = ica.find_bads_eog(raw)
+ica.exclude = eog_indices
+
+# barplot of IC component "EOG match" scores
+ica.plot_scores(eog_scores)
+
+# plot diagnostics
+ica.plot_properties(raw, picks=eog_indices)
+
+# plot ICs applied to raw data, with EOG matches highlighted
+ica.plot_sources(raw)
+
+# plot ICs applied to the averaged EOG epochs, with EOG matches highlighted
+ica.plot_sources(eog_evoked)
 
 ###############################################################################
-# These labels were used by the plotters and are added automatically
-# by artifact detection functions. You can also manually edit them to annotate
-# components.
+# Note that above we used :meth:`~mne.preprocessing.ICA.plot_sources` on both
+# the original :class:`~mne.io.Raw` instance and also on an
+# :class:`~mne.Evoked` instance of the extracted EOG artifacts. This can be
+# another way to confirm that :meth:`~mne.preprocessing.ICA.find_bads_eog` has
+# identified the correct components.
 #
-# Now let's see how we would modify our signals if we removed this component
-# from the data.
-ica.plot_overlay(eog_average, exclude=eog_inds, show=False)
-# red -> before, black -> after. Yes! We remove quite a lot!
+#
+# Using a simulated channel to select ICA components
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# If you don't have an EOG channel,
+# :meth:`~mne.preprocessing.ICA.find_bads_eog` has a ``ch_name`` parameter that
+# you can use as a proxy for EOG. You can use a single channel, or create a
+# bipolar reference from frontal EEG sensors and use that as virtual EOG
+# channel. This carries a risk however: you must hope that the frontal EEG
+# channels only reflect EOG and not brain dynamics in the prefrontal cortex (or
+# you must not care about those prefrontal signals).
+#
+# For ECG, it is easier: :meth:`~mne.preprocessing.ICA.find_bads_ecg` uses
+# cross-channel averaging to estimate a virtual ECG channel, so it is usually
+# not necessary to pass a specific channel name.
```

Comment:
As far as I understood it, virtual ECG channel creation is only possible for MEG data: https://martinos.org/mne/stable/generated/mne.preprocessing.find_ecg_events.html